### PR TITLE
Fix potential Null pointer dereference in bench_httpclient.c

### PR DESCRIPTION
**Pull request by icy17**

### CODE REVIEW

File: test/bench_httpclient.c [hunk 1]
```diff
@@ -163,7 +163,9 @@ launch_request(void)
 	evutil_gettimeofday(&ri->started, NULL);
 
 	b = bufferevent_socket_new(base, sock, BEV_OPT_CLOSE_ON_FREE);
-
+	if (b == NULL) {
+		return -1;
```
Comment (azat) [inline test/bench_httpclient.c:167]: Should close the fd as well, and also let's improve the caller to print which request failed

**Comment by icy17** [comment]
I just add cleanup, but I'm not sure how to "print which request failed" :(

**Review by azat — CHANGES_REQUESTED** [review]
>I just add cleanup, but I'm not sure how to "print which request failed" :(

Replace `perror("launch")` with `fprintf(stderr, "Cannot launch %i requet\n", i);` and can you also please add an error path into main

**Comment by azat** [comment]
Merged https://github.com/libevent/libevent/pull/1676 with co-authorship instead. Thank you for you work.

**Pull request closed without merging**